test/image/mocks/z-new_tickmode_sync.json

```diff
@@ -0,0 +1,91 @@
+{
+   "data": [
+      {
+         "name": "Apples",
+         "type": "bar",
+         "x": [
+            "Jan",
+            "Feb",
+            "Mar",
+            "Apr",
+            "May"
+         ],
+         "y": [
+            232,
+            2506,
+            470,
+            1864,
+            -190
+         ]
+      },
+      {
+         "name": "Oranges",
+         "type": "scatter",
+         "x": [
+            "A",
+            "B",
+            "C"
+        ],
+        "y": [
+            0,
+            0.5,
+            1
+        ],
+         "yaxis": "y2",
+         "xaxis": "x2"
+      }
+   ],
+   "layout": {
+      "margin": {
+         "t": 40,
+         "r": 70,
+         "b": 40,
+         "l": 70
+      },
+      "width": 700,
+      "showlegend": false,
+      "xaxis2": {
+         "anchor": "y2",
+         "side": "top",
+         "overlaying": "x",
+         "tickmode": "sync",
+         "domain": [
+            0.52,
+            1
+         ]
+      },
```

Comment:
Also let's describe `xaxis` and test *period* `ticklabelmode` on *date* axis e.g.
```json
      "xaxis": {
         "ticklen": 10,
         "tickwidth": 3,
         "ticklabelmode": "period"
      },
```

Reply:
We should also enable grids for *date* axes. So the `axis` should look like:
```json
      "xaxis": {
         "showgrid": true,
         "ticklen": 10,
         "tickwidth": 3,
         "ticklabelmode": "period"
      },
```